rrmngmnt/nmcli.py

```diff
@@ -0,0 +1,737 @@
+            ipv6_method (str): setting method.
+                Available methods: auto, dhcp, disabled, ignore, link-local,
+                manual, shared.
+            mtu (int): MTU to set for the connection.
+
+        Returns:
+            str: an nmcli connection add command with the passed in optional
+                parameters.
+        """
+        command = ""
+        if mtu:
+            command += " mtu {mtu}".format(mtu=mtu)
+        if ipv4_addr:
+            if not self._get_ip_version(ip_address=ipv4_addr) == 4:
+                raise InvalidIPException(ip=ipv4_addr)
+        if ipv4_gw:
+            if not self._get_ip_version(ip_address=ipv4_gw) == 4:
+                raise InvalidIPException(ip=ipv4_gw)
+        if ipv6_addr:
+            if not self._get_ip_version(ip_address=ipv6_addr) == 6:
+                raise InvalidIPException(ip=ipv6_addr)
+        if ipv6_gw:
+            if not self._get_ip_version(ip_address=ipv6_gw) == 6:
+                raise InvalidIPException(ip=ipv6_gw)
+        if ipv4_method:
+            command += " " + self._generate_ip_options(
+                ip_method=ipv4_method, address=ipv4_addr, gateway=ipv4_gw, version=4  # noqa: E501
+            )
+        if ipv6_method:
+            command += " " + self._generate_ip_options(
+                ip_method=ipv6_method, address=ipv6_addr, gateway=ipv6_gw, version=6  # noqa: E501
+            )
+        return command
+
+    @staticmethod
+    def _generate_common_options(con_type, con_name, ifname, auto_connect, save):  # noqa: E501
+        """
+        Generates a string containing common options for the nmcli tool.
+
+        Args:
+            con_type (str): the connection type.
+            con_name (str): the created connection's name.
+            ifname (str): the interface name to use.
+            auto_connect (bool): True to connect automatically, or False for
+                manual.
+            save (bool): True to persist the connection, or False.
+
+        Returns:
+            str: a common options string.
+        """
+        return COMMON_OPTIONS.format(
+            type=con_type,
+            con_name=con_name,
+            ifname=ifname,
+            auto_connect="yes" if auto_connect else "no",
+            save="yes" if save else "no",
+        )
+
+    @staticmethod
+    def _generate_ip_options(ip_method, address, gateway, version):
+        """
+        Generates a string containing ip options for the nmcli tool.
+
+        Args:
+            ip_method (str): setting method.
+                Available methods: auto, disabled, link-local, manual, shared.
+            address (str): a static address.
+            gateway (str): a gateway address.
+            version (int): IP version.
+
+        Returns:
+            str: an ip options string.
+        """
+        ip_options = ""
+
+        if ip_method:
+            ip_options += (
+                IPV4_METHOD.format(method=ip_method)
```

Comment:
If command are used only once please just use them, no need to save them as variables.